src/OpenSpaceToolkit/Physics/Coordinate/Spherical/LLA.cpp

```diff
@@ -109,6 +112,20 @@ Length LLA::calculateDistanceTo(
     return LLA::DistanceBetween(*this, aLLA, anEllipsoidEquatorialRadius, anEllipsoidFlattening);
 }
 
+Length LLA::calculateDistanceTo(const LLA& aLLA) const
+{
+    if (!Environment::AccessInstance()->hasCentralCelestial())
+    {
+        throw ostk::core::error::runtime::Undefined("Central celestial");
+    }
+
+    return this->calculateDistanceTo(
+        aLLA,
+        Environment::AccessInstance()->accessCentralCelestial()->getEquatorialRadius(),
+        Environment::AccessInstance()->accessCentralCelestial()->getFlattening()
+    );
+}
```

Comment:
Maybe I'm lacking contest here, my only feedback is that the fact that this is measured w.r.t. to the global Environment's central body is a bit hidden. Maybe we can allow for an **optional ** Central Body or Environment argument:

```suggestion
Length LLA::calculateDistanceTo(const LLA& aLLA, const Shared<const Celestial> aCentralBody) const
{
    // If no Central Body default to global Environment central boody
    
    return this->calculateDistanceTo(
        aLLA,
        aCentralBody->getEquatorialRadius(),
        aCentralBody->getFlattening()
    );
}
```

Or 

```suggestion
Length LLA::calculateDistanceTo(const LLA& aLLA, const Shared<const Environment> anEnvironment) const
{
    // If no environment, default to global Environment

    return this->calculateDistanceTo(
        aLLA,
        anEnvironment->accessCentralCelestial()->getEquatorialRadius(),
        anEnvironment->accessCentralCelestial()->getFlattening()
    );
}
```

Reply:
I agree it's not obvious. I think we should update the OSTk README and explicitly explain how the new global environment singleton will be used across the repos. I will push a commit let me know.